### ISSUE TITLE
sum panics on overflow..

### ISSUE THREAD

**Issue by earney** [gluesql/gluesql#579]
```
gluesql> create table mytable (id int(8));
gluesql> insert into mytable values (100);
1 row inserted

gluesql> insert into mytable values (50);
1 row inserted

gluesql> select sum(id) from mytable;
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Value(BinaryOperationOverflow { lhs: I8(100), rhs: I8(50), operator: Add })', core/src/executor/aggregate/state.rs:62:72
```

**Comment by earney** [comment]
I assume if we get an overflow, we need to try to upcast it to the next largest supported data type (I8 to I64) before throwing an overflow?
I8(100) + I8(50) => I8(150)  (overflow) vs
I8(100 + I8(50) => I64(150)    **** right answer.. (no overflow).

**Comment by earney** [comment]
modifying try_add like so, seems to fix the issue:

```
 I8(rhs) => match lhs
                .checked_add(rhs) {
                    Some(x) => Ok(I8(x)),
                    None => self.try_add(&I64(rhs as i64)),
                },
```

**Comment by ever0de** [comment]
I think we need a discussion about which type to cast to.

**Comment by earney** [comment]
> I think we need a discussion about which type to cast to.

I agree..   I also plan on implementing i32 (after the i128 PR is merged), so we could cast it to an i32 instead if that is what people want.

As everyone knows, we could also reach the same limits for other data types, i64 for example.  If checked_add returns None, we could auto promote to the next largest data type.  This makes sense to me.
What do other think?   If we implement this, each time we hit an overflow, we could just promote to the next largest data type, and then only i128 would throw an overflow.  (this of course is ignoring the decimal data type).
